sdks/python/apache_beam/runners/interactive/extensions/apache-beam-jupyterlab-sidepanel/src/SidePanel.ts

```diff
@@ -35,24 +40,30 @@ import {
 export class SidePanel extends BoxPanel {
   constructor(
     manager: ServiceManager.IManager,
-    rendermime: IRenderMimeRegistry
+    rendermime: IRenderMimeRegistry,
+    pageType: string
   ) {
     super({
       direction: 'top-to-bottom',
       alignment: 'end'
     });
     this.id = 'apache-beam-jupyterlab-sidepanel';
-    this.title.label = 'Interactive Beam Inspector';
     this.title.closable = true;
-
     this._sessionContext = new SessionContext({
       sessionManager: manager.sessions,
       specsManager: manager.kernelspecs,
       name: 'Interactive Beam Inspector Session'
     });
 
-    this._inspector = new InteractiveInspectorWidget(this._sessionContext);
-    this.addWidget(this._inspector);
+    if (pageType === 'Inspector') {
+      this.title.label = 'Interactive Beam Inspector';
+      this._inspector = new InteractiveInspectorWidget(this._sessionContext);
+      this.addWidget(this._inspector);
+    } else if (pageType === 'Clusters') {
+      this.title.label = 'Interactive Beam Cluster Manager';
+      this._clusters = new ClustersWidget(this._sessionContext);
+      this.addWidget(this._clusters);
+    }
```

Comment:
It's a little weird to have the SidePanel know how to construct a widget based on a string parameter. In the future if we want to add more widgets that each construct using different parameters this code could get messy. Instead, I suggest giving the sidepanel the widget and title as parameters.

Reply:
That is a great point. Thanks for the suggestion! I now pass in the widget and the code is much cleaner.